include/seastar/core/shared_future.hh

```diff
@@ -184,53 +194,41 @@ private:
                     abort_source& as = e.timer->abort_source();
                    _peers.make_back_abortable(as);
                 }
-                if (_original_future._state.valid()) {
-                    // _original_future's result is forwarded to each peer.
-                    (void)_original_future.then_wrapped([s = this->shared_from_this()] (future_type&& f) mutable {
-                        s->resolve(std::move(f));
-                    });
-                }
                 return f;
-            } else if (_original_future.failed()) {
-                return future_type(exception_future_marker(), std::exception_ptr(_original_future._state.get_exception()));
+            } else if (this->_state.failed()) {
+                return future_type(exception_future_marker(), std::exception_ptr(this->_state.get_exception()));
             } else {
-                return future_type(ready_future_marker(), _original_future._state.get_value());
+                return future_type(ready_future_marker(), this->_state.get_value());
             }
         }
 
         future_type get_future(abort_source& as) noexcept {
```

Comment:
and this is why it's a mistake to push abort_source everywhere @kbr-scylla

Reply:
Please elaborate. You mean the code duplication?